tests/ocf-url_parse_leaking_relative/EPUB/content_001.xhtml

```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
+    <head>
+        <meta charset="utf-8"/>
+        <title>Parse a potentially leaking URL</title>
+        <script type="text/javascript">
+            const expected_path   = 'media/imgs/monastery.jpg';
+            const parent_relative = `../../../../../${expected_path}`;
+            const current_path    = 'EPUB/content_001.xhtml'
+            function test_url() {
+                try {
+                    document.getElementById('no_scripting').style.display = 'none';
+                    document.getElementById('scripting').style.display = 'block';
+
+                    // in some cases this URL contains search or fragment, this should be removed to get to the
+                    // 'pure' URL. This is done by cloning the document's URI and removing  
+                    // relevant portions of the full URL
+                    // (There are RS-s that add a long query string to the URL, hence the necessity of this step)
+                    const new_URL = new URL(document.documentURI);
+                    new_URL.search = '';  new_URL.hash = '';
+                    const documentURI = new_URL.toString();
+
+                    // This is, hopefully, a good representation of the container root URL: remove the
+                    // relative path of the current document
+                    const container_root_URL = documentURI.slice(0,documentURI.length - current_path.length);
```

Comment:
couldn't all this be replaced by:

```js
const container_root_URL = `new URL('../..', document.documentURI)
```

Reply:
What made me stop and do what I did was if the RS uses a URL for the document of the form:

```
file://path/to.epub#path=/EPUB/content.opf
```
or

```
jar:file:/path/to.epub!/EPUB/content.opf
```

(these are [your examples](https://github.com/w3c/epub-specs/issues/1374#issuecomment-847978623)😉)

But, on the other hand, if the parsing of `../..` go wrong, that means that RS is non-conform anyway... which is a problem with the RS then...

(And this may make the previous clean-up of the hash and search also unnecessary.)